src/main/java/uk/gov/hmcts/reform/unspec/launchdarkly/OnBoardingOrganisationControlService.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.unspec.launchdarkly;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.launchdarkly.sdk.LDValue;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.prd.model.Organisation;
+import uk.gov.hmcts.reform.unspec.service.OrganisationService;
+import uk.gov.hmcts.reform.unspec.service.UserService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class OnBoardingOrganisationControlService {
+
+    public static final String ORG_NOT_REGISTERED = "your organisation is %s, unfortunately that org "
+        + "is not part of pilot and therefore cannot use Civil Damages service";
+    
+    private final FeatureToggleService featureToggleService;
+    private final OrganisationService organisationService;
+    private final ObjectMapper objectMapper;
+    private final UserService userService;
+
+    public List<String> isOrganisationAllowed(String userBearer) {
+        List<String> errors = new ArrayList<>();
+        boolean isSystemUpdateUser = userService.getUserInfo(userBearer).getRoles().stream()
+            .anyMatch(r -> r.equals("caseworker-civil-systemupdate"));
+
+        if (isSystemUpdateUser) {
+            return errors;
+        }
+
+        Optional<Organisation> userOrganisation = organisationService.findOrganisation(userBearer);
+
+        try {
+            LDValue registeredFirms = featureToggleService.jsonValueFeature("registeredFirms");
+
+            OnboardedOrganisation onboardedOrganisation
+                = objectMapper.readValue(registeredFirms.toJsonString(), OnboardedOrganisation.class);
+
+            Boolean orgIsInMyHmcts = userOrganisation.map(userOrg -> hasSameOrganisation(
+                onboardedOrganisation,
+                userOrg
+            )).orElse(false);
+
+            if (!orgIsInMyHmcts) {
+                errors.add(String.format(
+                    ORG_NOT_REGISTERED,
+                    userOrganisation.map(Organisation::getName).orElse("UnRegistered")
+                ));
+            }
+        } catch (JsonProcessingException jsonProcessingException) {
+            log.error("invalid json list of registered firms in launch darkly config");
+        }
```

Comment:
We don't have to do that kind of check in the code, LaunchDarkly can do it for us. If we send organisation in `LDUser` we can add targeting rule in LaunchDarkly. It'd make the code much simpler, it'd be easier to maintain than JSON value (there is an option to add `is one of` rule) and of course less error-prone.